FreeAPS/Sources/APS/FetchTreatmentsManager.swift

```diff
@@ -22,18 +23,23 @@ final class BaseFetchTreatmentsManager: FetchTreatmentsManager, Injectable {
     private func subscribe() {
         timer.publisher
             .receive(on: processQueue)
-            .flatMap { _ -> AnyPublisher<([CarbsEntry], [TempTarget]), Never> in
+            .flatMap { _ -> AnyPublisher<(Date, [CarbsEntry], [TempTarget], [CarbsEntry]), Never> in
                 debug(.nightscout, "FetchTreatmentsManager heartbeat")
                 debug(.nightscout, "Start fetching carbs and temptargets")
-                return Publishers.CombineLatest(
+                return Publishers.CombineLatest4(
+                    Just(self.carbsStorage.syncDate()),
                     self.nightscoutManager.fetchCarbs(),
-                    self.nightscoutManager.fetchTempTargets()
+                    self.nightscoutManager.fetchTempTargets(),
+                    self.healthKitManager.fetch()
                 ).eraseToAnyPublisher()
             }
-            .sink { carbs, targets in
-                let filteredCarbs = carbs.filter { !($0.enteredBy?.contains(CarbsEntry.manual) ?? false) }
+            .sink { syncDate, carbs, targets, carbsFromHealth in
+                let allCarbs = carbs + carbsFromHealth
+                let filteredCarbs = allCarbs.filter { $0.createdAt > syncDate }
                 if filteredCarbs.isNotEmpty {
+                    debug(.nightscout, "Filtered carbs: \(filteredCarbs)")
                     self.carbsStorage.storeCarbs(filteredCarbs)
+                    self.healthKitManager.saveIfNeeded(carbs: filteredCarbs)
```

Comment:
This code save all data (wich contains data from health) to health. After that we have dublicates in AH.
Please add
`let carbsForHealth = allCarbs.filter { !carbsFromHealth.contains($0) }`
and save only filtered data
self.healthKitManager.saveIfNeeded(carbs: carbsForHealth)